server/rest/src/main/java/org/infinispan/rest/resources/ProtobufResource.java

```diff
@@ -0,0 +1,233 @@
+               .filter(cacheEntry -> !((String)cacheEntry.getKey()).endsWith(ProtobufMetadataManager.ERRORS_KEY_SUFFIX))
+               .map(cacheEntry -> {
+                  String error = (String) cache
+                        .get(cacheEntry.getKey() + ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX);
+                  ProtoSchema protoSchema = new ProtoSchema();
+                  protoSchema.name = cacheEntry.getKey().toString();
+                  if (error != null) {
+                     protoSchema.error = createErrorContent(protoSchema.name, error);
+                  }
+                  return protoSchema;
+               })
+                .collectInto(new HashSet<ProtoSchema>(), Set::add)
+                .map(protoSchemas -> {
+                  try {
+                     List<ProtoSchema> sortedCacheInfos = protoSchemas.stream()
+                           .sorted(Comparator.comparing(s -> s.name))
+                           .collect(Collectors.toList());
+                     byte[] bytes = objectMapper.writeValueAsBytes(sortedCacheInfos);
+                     responseBuilder.contentType(APPLICATION_JSON).entity(bytes);
+                  } catch (JsonProcessingException e) {
+                     responseBuilder.status(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+                  }
+                  return (RestResponse) responseBuilder.build();
+               })
+                .toCompletionStage()
+      , invocationHelper.getExecutor())
+            .thenCompose(Function.identity());
+   }
+
+   private CompletionStage<RestResponse> addOrEditSchema(RestRequest request, boolean create) {
+
+      String schemaName = checkMandatorySchemaName(request);
+
+      ContentSource contents = request.contents();
+      if (contents == null) throw new NoDataFoundException("Schema data not sent in the request");
+
+      AdvancedCache<Object, Object> cache = invocationHelper.getRestCacheManager()
+            .getCache(ProtobufMetadataManager.PROTOBUF_METADATA_CACHE_NAME, request);
+
+
+      NettyRestResponse.Builder builder = new NettyRestResponse.Builder();
+
+      CompletableFuture<Object> putSchema;
+      if (create) {
+         putSchema = cache.putIfAbsentAsync(schemaName, contents.asString()).thenApply(result -> {
+            if (result == null) {
+               builder.status(HttpResponseStatus.CREATED);
+            } else {
+               builder.status(HttpResponseStatus.CONFLICT);
+            }
+            return result;
+         });
+      } else {
+         putSchema = cache.replaceAsync(schemaName, contents.asString()).thenApply(result -> {
+            if (result == null) {
+               builder.status(HttpResponseStatus.NOT_FOUND);
+            } else {
+               builder.status(HttpResponseStatus.OK);
+            }
+            return result;
+         });
+      }
+
+      return putSchema
+            .thenCompose(r -> {
+               if(isOkOrCreated(builder)) {
```

Comment:
Nitpick: spacing on if blocks are missing here and a few below.

Reply:
ok